website_twitter_no_ext_links/models/twitter_feed.py

```diff
@@ -0,0 +1,147 @@
+                args)
+            request = urllib2.Request(url)
+            request.add_header(
+              "Authorization",  "Bearer %s" %  token
+            )
+            request.add_header(
+                'content-type',  'application/x-www-form-urlencoded;charset=UTF-8'
+            )
+            try:
+                response = urllib2.urlopen(request)
+            except urllib2.HTTPError as e:
+                print e.code
+                print e.reason
+                print e.read()
+            userposts = response.read()
+            userposts_dict = json.loads(userposts)
+            if userposts_dict:
+                feed.clear_tweets()
+                for item in userposts_dict:
+                    tweet_formatted_text = item['text']
+                    at_users = re.findall('(?<=@)\w+', tweet_formatted_text)
+                    hashtags = re.findall('(?<=#)\w+', tweet_formatted_text)
+                    for at_user in at_users:
+                        tweet_formatted_text = tweet_formatted_text.replace(
+                            '@'+at_user,
+                            '<span color="' + feed.theme.twitter_at_color + '">@' + at_user + '</span>'
+                        )
+                    for hashtag in hashtags:
+                        tweet_formatted_text = tweet_formatted_text.replace(
+                            '#'+hashtag,
+                            '<span color="' + feed.theme.twitter_hashtag_color + '">#' + hashtag + '</span>'
+                        )
+                    tweet_dict = {
+                        'text': tweet_formatted_text,
+                        'author': item['user']['name'],
+                        'tweet_create_date': item['created_at'],
+                        'tweet_id': item['id'],
+                        'feed_id': feed.id,
+                        }
+                    self.env['twitter.tweet'].create(tweet_dict)
+                    for entity in item['entities']['user_mentions']:
+                        entitydict = {
+                            'twitter_user_id': entity['id'],
+                            'twitter_user_name': entity['name'],
+                            'twitter_user_screen_name': entity['screen_name'],
+                            'mentions': 1,
+                            'feed_id': feed.id,
+                            }
+                        exists = self.env['twitter.user'].search(
+                            [('twitter_user_id', '=', entity['id']),
+                             ('feed_id', '=', self.id)
+                            ]
+                        )
+                        if exists:
+                            exists[0].write({'mentions': exists.mentions +1})
+                        else:
+                            self.env['twitter.user'].create(entitydict)
+
+    def clear_tweets(self):
+        #Proposal maybe not delete but "archive"
```

Comment:
that's why god gave us the `active` flag. But wouldn't this lead to data duplication?